README.txt
// $Id: README.txt,v 1.7.2.1.2.2 2009/07/23 20:56:45 xamanu Exp $

DESCRIPTION
-----------
Timeline is a DHTML-based AJAXy widget for visualizing temporal information.
It works in a similar way as Google Maps but for time-based events.

This Drupal module is based on a JavaScript widget developed by the MIT
SIMILE project. Several demo timelines demonstrating the full range of 
capabilities for this widget are available on the SIMILE project site at:
- http://code.google.com/p/simile-widgets/

This module is a style plugin for the Views module that represents date 
information of nodes on horizontal or vertical timeline display.
- http://drupal.org/project/views

The timeline provides support for the Date module. CCK date fields can be used
to display all kinds of information on the timeline.
- http://drupal.org/project/cck
- http://drupal.org/project/date

The SIMILE Timeline (and Ajax) libraries can be used from the SIMILE server 
or can be downloaded and installed into your (local) Drupal installation using 
the libraries module. (See INSTALL.txt for more information). By default, the 
SIMILE Timeline (and Ajax) libraries can be used from the SIMILE server.
- http://drupal.org/project/libraries

A view is provided by default to get you started quickly: 
* timeline_nodes: plots any nodes by their created date

Timeline events with only a start date are displayed as points on the
timeline view, whilst events with both a start and date are displayed as
continuous colored blocks (not dissimilar to Gantt chart entries).


REQUIREMENTS
------------
Please refer to the accompanying INSTALL.txt file.


INSTALLATION
------------
Please refer to the accompanying INSTALL.txt file.


USAGE
-----
After installing the module, do the following to create a new timeline view:

1. Make sure the `views' and `views_ui' modules are installed and enabled.
2. Go to administer >> views >> add.
3. Make a new view that is of the view type "Node"
4. Under Fields, add that fields that you would like to have populate the title, 
   start and end dates and body of the timeline popup.
   Note: Make sure that you are using exclusive date fields. Don't choose "Show from and to date"
5. Set any other views settings, such as the maximum number of nodes to
   show, filters to select only certain nodes or dates, etc. You'll likely
   want to disable the view's pager option in order to get all events to
   show up on the timeline.
6. Under "Basic settings" >> "Style" select the option "Timeline".
7. Configure the Timeline display (click the gear next to the style selector). 
   Set the orientation, display and field settings that control how the 
   timeline looks and behaves. 
   Please take special care when selecting the "field usage". These fields let you map
   the fields selected in your view to the fields used by the timeline.
8. Save your view and navigate to it to test out the timeline. In case you
   don't see the timeline at all, that most likely indicates the view
   selection criteria didn't result in any nodes to be displayed.


CREDITS
---------------
Version 2.x 
Maintained by Felix Delattre [1] with kindly support of Jeff Miccolis (Development Seed) [2].
The included Fancy Theme is based on work by Shawn Kindling [3]

Version 1.x 
Originally created by David Donohue [4] and further developed by Arto Bendiken [5] and 
Jeff Miccolis (Development Seed) [2].


Both timelines are based on the Timeline developed by the SIMILE project [6]


[1] Felix Delattre -> xamanu -> http://drupal.org/user/359937
[2] Jeff Miccolis -> jmiccolis -> http://drupal.org/user/31731
[3] Shawn Kindling -> http://keelingdesign.com/theming_a_simile_timeline
[4] David Donohue -> dado -> http://drupal.org/user/16745
[5] Art Bendiken -> arto -> http://drupal.org/user/26089
[6] SIMILE project -> http://code.google.com/p/simile-widgets/


TROUBLESHOOTING
---------------
To post bug reports or feature requests please use the following issue
trackers for the Drupal module and the MIT SIMILE widget, respectively:
- http://drupal.org/project/issues/timeline
- http://code.google.com/p/simile-widgets/issues/list


CHANGELOG.txt
// $Id: CHANGELOG.txt,v 1.2.4.1 2009/07/23 18:21:23 xamanu Exp $

2009/07/23 - 6.x-2.0
----------------------
- Partially rewritten
- Updated to Simile Timeline 2.3.0
- Extended visualization functions from Simile Timeline 2.x
- Object based architecture for timeline settings and events
- Better views 2 integration
- Grouping with color coding.
- Respecting Drupal's js behaviors
- Support for individual themes. Providing two new themes.
- Improved support for Date module
- Depending on libraries module for local installation.
- Dropped support for events module.
- Supporting json only.
- Avoiding js callback to certain url.
- #121377 Arbitrary/custom initial focus.


2008/07/05 - 5.x-1.0
----------------------
- Fixed various bugs related to timeline controls. (Thanks to franskuipers)
- #165634 Fixed link array format in timeline_link.
- #135658 Patch by KarenS - Adds date module support and end date auto-detection. 
          You may need to update your views.
- #99552: Update for Drupal 5.x

2006/12/09 - 4.7.x-1.0
----------------------
- Added an option for specifying which event/date the timeline views should
  initially be centered on after loading.
- Fixed widget creation so that the 'Loading...' bubble is displayed.
- #94031: Allow using custom CCK fields as the start/end date, title and
          description of timeline events (kudos to John Stalcup).
- #79472: Fix asynchronous timeline loading problems occuring in IE6 (kudos
          to John Stalcup and Erwin Lion).
- #91081: Corrected the timeline popup node links (kudos to Eric Charikane).
- #90660: Prevented other modules (e.g. devel.module) from intefering with
          timeline JSON or XML feed output (kudos to Steve Yelvington).
- #90272: Extended the module's installation and usage instructions, and
          reorganized the code for better extensibility.
- #90272: Implemented an installation hook which verifies that the required
          Views module is installed and enabled, and outputs a warning in
          case it is not.
- #90272: Implemented JSON output and added a setting for specifying whether
          to use JSON or XML for loading the timeline data feed.
- #90272: Simplified the XML output code by making use Drupal's built-in
          function format_xml_elements().
- #90272: Changed dates/times in data feed to use the ISO-8601 format, and
          added explicit date format specifier to data feeds.
- #90272: Added a setting for specifying the base URL to load the timeline
          JavaScript widget from.
- #90272: Added a setting and cache control implementation for specifying the
          client-side cache lifetime for timeline data feeds.
- #90272: Changed timeline data feed loading to use a relative, instead of
          absolute, URL to counteract problems when accessing a site via
          HTTP/HTTPS both.
- #90272: Added an explicit UTF-8 character set specifier to the HTTP
          Content-Type header of the data feeds.
- #90272: Implemented some initial Simpletest unit tests for verifying the
          well-formedness of the data feeds provided by the timeline module.
- #90272: Implemented a best-effort workaround for the widget's timezone
          display bugs, claiming dates are GMT while they're in fact
          explicitly adjusted to the current user's timezone.
- #90272: Added a wrapper DIV element around timeline DIV for easier control
          over theming, and added a visible border around the timeline
          widget for stylistic effect.
- #90272: Changed the default timeline CSS class name and width, and set the
          timeline XHTML element ID to the name of the current timeline view
          in order to facilitate custom theming.
- #90272: Separated timeline creation JavaScript code out of the module and
          into a file called timeline.js; also added an initial timeline.css
          stylesheet file.
- #90272: Successfully tested vertical timeline views (with newest version
          of the timeline widget).


test.js

// needs basepath. and it is not the correct location
//Timeline_ajax_url='/sites/all/libraries/simile_timeline/timeline_ajax/simile-ajax-api.js';
Timeline_urlPrefix="/sites/all/libraries/simile_timeline/timeline_js/";

js/timeline.js
Drupal.timeline = {	
  timerID: null,

  getTheme: function(theme, timeline_start, timeline_end, bubble_max_height, bubble_width) {
    var theme = Timeline[theme].create();
    theme.event.bubble.width = parseInt(bubble_width); // px
    theme.event.bubble.maxHeight = parseInt(bubble_max_height);
    //theme.event.label.width = parseInt(bubble_width);  // don't know what for -fd 
    return theme;
  },

  createWidget: function(args) {
  	args = eval(args);
    var theme = Drupal.timeline.getTheme(args.theme, args.timeline_start, args.timeline_end, args.bubble_max_height, args.bubble_width);
    var bandInfos = [];
    var Sources = [];

    var width = 100;
    if (args.bands.length === 1 && args.summary !== '') {
      width = width - 25;
    }
    else{
      width = 100 / args.bands.length;
    }
    
    $.each(args.bands, function(i, v) {
      var eventSource = new Timeline.DefaultEventSource();
      Sources.push(eventSource);
      bandInfos.push(
        Timeline.createBandInfo({
          width:          width + '%',
          intervalUnit:   Timeline.DateTime[v],
          intervalPixels: 200,
          eventSource:    eventSource,
          date:           args.initial_focus,
          timeZone:       args.timezone,
          theme:          theme
        })
      );
      
      if (bandInfos.length > 1) {
        bandInfos[bandInfos.length-1].syncWith = 0;
      }
    });
    
   if (args.bands.length === 1 && args.summary !== '') {
      bandInfos.push(
        Timeline.createBandInfo({
          width:          '25%',
          intervalUnit:   Timeline.DateTime[args.summary],
          intervalPixels: 200,
          eventSource:    Sources[0],
          date:           args.initial_focus,
          timeZone:       args.timezone,
          theme:          theme,
          overview:       true,
          trackHeight:    0.5,
          trackGap:       0.2
        })
      );

      bandInfos[1].syncWith = 0;
      bandInfos[1].highlight = true;
    }
   
   
   var timeline = Timeline.create(document.getElementById(args.id), bandInfos);

   
   var url = document.location.href;
   //timeline_sets = eval("(" + timeline_sets + ")");
   $.each(args.events.events, function(i, v) {
          Sources[i].loadJSON(args.events, url);
   });
  }
};

Drupal.timeline.controls = {
  setup: function(div, timeline, filterLabel, highlightLabel, clearButton) {
    var bandIndices = [0, 1];
    var theme = Drupal.timeline.getTheme();

    var table = document.createElement('table');
    var tr = table.insertRow(0);
    var td = tr.insertCell(0);
    td.innerHTML = filterLabel;

    td = tr.insertCell(1);
    td.innerHTML = highlightLabel;

    var handler = function(elmt, evt, target) {
      if (Drupal.timeline.TimerID !== null) {
        window.clearTimeout(Drupal.timeline.TimerID);
      }
      timelineTimerID = window.setTimeout(function() {
        Drupal.timeline.controls.filter(timeline, bandIndices, table);
      }, 300);
    };

    tr = table.insertRow(1);
    tr.style.verticalAlign = 'top';

    td = tr.insertCell(0);

    var input = document.createElement('input');
    input.type = 'text';
    Timeline.DOM.registerEvent(input, 'keypress', handler);
    td.appendChild(input);

    for (var i = 0; i < theme.event.highlightColors.length; i++) {
      td = tr.insertCell(i + 1);

      input = document.createElement('input');
      input.type = 'text';
      Timeline.DOM.registerEvent(input, 'keypress', handler);
      td.appendChild(input);

      var divColor = document.createElement('div');
      divColor.style.height = '0.5em';
      divColor.style.background = theme.event.highlightColors[i];
      td.appendChild(divColor);
    }

    td = tr.insertCell(tr.cells.length);
    var button = document.createElement('button');
    button.innerHTML = clearButton;
    Timeline.DOM.registerEvent(button, 'click', function() {
      Drupal.timeline.controls.clear(timeline, bandIndices, table);
    });
    td.appendChild(button);

    document.getElementById(div).appendChild(table);
  },

  cleanString: function(s) {
    return s.replace(/^\s+/, '').replace(/\s+$/, '');
  },

  filter: function(timeline, bandIndices, table) {
    Drupal.timeline.TimerID = null;

    var tr = table.rows[1];
    var text = Drupal.timeline.controls.cleanString(tr.cells[0].firstChild.value);

    var filterMatcher = null;
    if (text.length > 0) {
      var regex = new RegExp(text, 'i');
      filterMatcher = function(evt) {
        return regex.test(evt.getText()) || regex.test(evt.getDescription());
      };
    }

    var regexes = [];
    var hasHighlights = false;
    for (var x = 1; x < tr.cells.length - 1; x++) {
      var input = tr.cells[x].firstChild;
      var text2 = Drupal.timeline.controls.cleanString(input.value);
      if (text2.length > 0) {
        hasHighlights = true;
        regexes.push(new RegExp(text2, 'i'));
      }
      else {
        regexes.push(null);
      }
    }

    var highlightMatcher = hasHighlights ? function(evt) {
      var text = evt.getText();
      var description = evt.getDescription();
      for (var x = 0; x < regexes.length; x++) {
        var regex = regexes[x];
        if (regex !== null && (regex.test(text) || regex.test(description))) {
          return x;
        }
      }
      return -1;
    } : null;

    for (var i = 0; i < bandIndices.length; i++) {
      var bandIndex = bandIndices[i];
      timeline.getBand(bandIndex).getEventPainter().setFilterMatcher(filterMatcher);
      timeline.getBand(bandIndex).getEventPainter().setHighlightMatcher(highlightMatcher);
    }
    timeline.paint();
  },

  clear: function(timeline, bandIndices, table) {
    var tr = table.rows[1];
    for (var x = 0; x < tr.cells.length - 1; x++) {
      tr.cells[x].firstChild.value = '';
    }
    for (var i = 0; i < bandIndices.length; i++) {
      var bandIndex = bandIndices[i];
      timeline.getBand(bandIndex).getEventPainter().setFilterMatcher(null);
      timeline.getBand(bandIndex).getEventPainter().setHighlightMatcher(null);
    }
    timeline.paint();
  }
};

Drupal.behaviors.timeline = function(context) {
  if(Timeline && Drupal.settings.timeline) {
    $.each(Drupal.settings.timeline, function(i, v) {
      $('.' + i + '-wrapper:not(.timeline-processed)', context).each(function() {
        $(this).addClass('timeline-processed');
        Drupal.timeline.createWidget(v);
      });
    });
  } 
  /*if ($controls) {
    $texts = implode(', ', array_map('drupal_to_js', array(t('Filter:'), t('Highlight:'), t('Clear All'))));
    $script = 'var timeline = '. $script ." Drupal.timeline.controls.setup('{$timeline_array->id}-controls', timeline, $texts);";
  }*/
};

themes/FancyTheme/FancyTheme.js
/*==================================================
 *  Fancy Theme
 *==================================================
 */



Timeline.FancyTheme = new Object();

Timeline.FancyTheme.implementations = [];

Timeline.FancyTheme.create = function(locale) {
    if (locale == null) {
        locale = Timeline.getDefaultLocale();
    }
    
    var f = Timeline.FancyTheme.implementations[locale];
    if (f == null) {
        f = Timeline.FancyTheme._Impl;
    }
    return new f();
};

Timeline.FancyTheme._Impl = function() {
    this.firstDayOfWeek = 0; // Sunday
          
    // Note: Many styles previously set here are now set using CSS
    //       The comments indicate settings controlled by CSS, not
    //       lines to be un-commented.
    //
    //
    // Attributes autoWidth, autoWidthAnimationTime, timeline_start 
    // and timeline_stop must be set on the first band's theme.
    // The other attributes can be set differently for each 
    // band by using different themes for the bands.
    this.autoWidth = false; // Should the Timeline automatically grow itself, as
                            // needed when too many events for the available width
                            // are painted on the visible part of the Timeline?
    this.autoWidthAnimationTime = 500; // mSec
    this.timeline_start = null; // Setting a date, eg new Date(Date.UTC(2008,0,17,20,00,00,0)) will prevent the
                                // Timeline from being moved to anytime before the date.
    this.timeline_stop = null;  // Use for setting a maximum date. The Timeline will not be able 
                                // to be moved to anytime after this date.
    this.ether = {
        backgroundColors: [
        //    "#F00",
        //    "#DDD",
        //    "#CCC",
        //    "#AAA"
        ],
        highlightColor:     "#FFF",
        highlightOpacity:   20,
        interval: {
            line: {
                show:       true,
                opacity:    25
               // color:      "#aaa",
            },
            weekend: {
                opacity:    30,
                  color:    "#FFFFE0",
            },
            marker: {
                hAlign:     "Bottom",
                hBottomStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-bottom";
                },

                hBottomEmphasizedStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-bottom-emphasized";
                },

                hTopStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-top";
                },

                hTopEmphasizedStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-top-emphasized";
                },
                vAlign:     "Right",
                vRightStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-right";
                },

                vRightEmphasizedStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-right-emphasized";
                },

                vLeftStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-left";
                },

                vLeftEmphasizedStyler:function(elmt) {
                    elmt.className = "timeline-ether-marker-left-emphasized";
                }                
            }
        }
    };
    
    this.event = {
        track: {
                   height: 10, // px. You will need to change the track
                               //     height if you change the tape height.
                      gap:  5, // px. Gap between tracks
                   offset:  10, // px. top margin above tapes
          autoWidthMargin:  1.5
          /* autoWidthMargin is only used if autoWidth (see above) is true.
             The autoWidthMargin setting is used to set how close the bottom of the
             lowest track is to the edge of the band's div. The units are total track
             width (tape + label + gap). A min of 0.5 is suggested. Use this setting to
             move the bottom track's tapes above the axis markers, if needed for your
             Timeline.
          */
        },
        overviewTrack: {
                  offset: 20, // px -- top margin above tapes 
              tickHeight:  6, // px
                  height:  2, // px
                     gap:  1, // px
         autoWidthMargin:  5 // This attribute is only used if autoWidth (see above) is true.
        },
        tape: {
            height:         4 // px. For thicker tapes, remember to change track height too.
        },
        instant: {
                           icon: Timeline.urlPrefix + "../../../modules/timeline/themes/FancyTheme/images/circle.png", 
                                 // default icon. Icon can also be specified per event
                      iconWidth: 10,
                     iconHeight: 10,
               impreciseOpacity: 20, // opacity of the tape when durationEvent is false
            impreciseIconMargin: 3,   // A tape and an icon are painted for imprecise instant
                                     // events. This attribute is the margin between the
                                     // bottom of the tape and the top of the icon in that
                                     // case.
            color:             "#333366",
            impreciseColor:    "#333366"
        },
        duration: {
            impreciseOpacity: 20, // tape opacity for imprecise part of duration events
            opacity:        20, // color opacity, not text opacity
            color:            "#333366",
            impreciseColor:   "#333366"
        },
        label: {
            backgroundOpacity: 50,// only used in detailed painter
               offsetFromLine:  3, // px left margin amount from icon's right edge
              backgroundColor:   "white",
              lineColor:         "#58A0DC"
        },
        highlightColors: [  // Use with getEventPainter().setHighlightMatcher
                            // See webapp/examples/examples.js
            "#FFFF00",
            "#FFC000",
            "#FF0000",
            "#0000FF"
        ],
        highlightLabelBackground: false, // When highlighting an event, also change the event's label background?
        bubble: {
            width:          350, // px
            maxHeight:        0, // px Maximum height of bubbles. 0 means no max height. 
                                 // scrollbar will be added for taller bubbles
            titleStyler: function(elmt) {
                elmt.className = "timeline-event-bubble-title";
            },
            bodyStyler: function(elmt) {
                elmt.className = "timeline-event-bubble-body";
            },
            imageStyler: function(elmt) {
                elmt.className = "timeline-event-bubble-image";
            },
            wikiStyler: function(elmt) {
                elmt.className = "timeline-event-bubble-wiki";
            },
            timeStyler: function(elmt) {
                elmt.className = "timeline-event-bubble-time";
            }
        }
    };
    
    this.mouseWheel = 'scroll'; // 'default', 'zoom', 'scroll'
};

themes/darkTheme.js
/*==================================================
 *  Classic Theme
 *==================================================
 */



Timeline.ClassicTheme = new Object();

Timeline.ClassicTheme.implementations = [];

Timeline.ClassicTheme.create = function(locale) {
    if (locale == null) {
        locale = Timeline.getDefaultLocale();
    }
    
    var f = Timeline.ClassicTheme.implementations[locale];
    if (f == null) {
        f = Timeline.ClassicTheme._Impl;
    }
    return new f();
};

Timeline.ClassicTheme._Impl = function() {
    this.firstDayOfWeek = 0; // Sunday
          
    // Note: Many styles previously set here are now set using CSS
    //       The comments indicate settings controlled by CSS, not
    //       lines to be un-commented.
    //
    //
    // Attributes autoWidth, autoWidthAnimationTime, timeline_start 
    // and timeline_stop must be set on the first band's theme.
    // The other attributes can be set differently for each 
    // band by using different themes for the bands.
    this.autoWidth = false; // Should the Timeline automatically grow itself, as
                            // needed when too many events for the available width
                            // are painted on the visible part of the Timeline?
    this.autoWidthAnimationTime = 500; // mSec
    this.timeline_start = null; // Setting a date, eg new Date(Date.UTC(2008,0,17,20,00,00,0)) will prevent the
                                // Timeline from being moved to anytime before the date.
    this.timeline_stop = null;  // Use for setting a maximum date. The Timeline will not be able 
                                // to be moved to anytime after this date.
    this.ether = {
        backgroundColors: [
        //    "#EEE",
        //    "#DDD",
        //    "#CCC",
        //    "#AAA"
        ],
     //   highlightColor:     "white",
        highlightOpacity:   50,
        interval: {
            line: {
                show:       true,
                opacity:    25
               // color:      "#aaa",
            },
            weekend: {
                opacity:    50
              //  color:      "#FFFFE0",
            },
            marker: {
                hAlign:     "Bottom",
                vAlign:     "Right"
                                        /*
                hBottomStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-bottom";
                },
                hBottomEmphasizedStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-bottom-emphasized";
                },
                hTopStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-top";
                },
                hTopEmphasizedStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-top-emphasized";
                },
                */
                                        
                    
               /*
                                  vRightStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-right";
                },
                vRightEmphasizedStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-right-emphasized";
                },
                vLeftStyler: function(elmt) {
                    elmt.className = "timeline-ether-marker-left";
                },
                vLeftEmphasizedStyler:function(elmt) {
                    elmt.className = "timeline-ether-marker-left-emphasized";
                }
                */
            }
        }
    };
    
    this.event = {
        track: {
                   height: 10, // px. You will need to change the track
                               //     height if you change the tape height.
                      gap:  2, // px. Gap between tracks
                   offset:  2, // px. top margin above tapes
          autoWidthMargin:  1.5
          /* autoWidthMargin is only used if autoWidth (see above) is true.
             The autoWidthMargin setting is used to set how close the bottom of the
             lowest track is to the edge of the band's div. The units are total track
             width (tape + label + gap). A min of 0.5 is suggested. Use this setting to
             move the bottom track's tapes above the axis markers, if needed for your
             Timeline.
          */
        },
        overviewTrack: {
                  offset: 20, // px -- top margin above tapes 
              tickHeight:  6, // px
                  height:  2, // px
                     gap:  1, // px
         autoWidthMargin:  5 // This attribute is only used if autoWidth (see above) is true.
        },
        tape: {
            height:         4 // px. For thicker tapes, remember to change track height too.
        },
        instant: {
                           icon: Timeline.urlPrefix + "images/dull-blue-circle.png", 
                                 // default icon. Icon can also be specified per event
                      iconWidth: 10,
                     iconHeight: 10,
               impreciseOpacity: 20, // opacity of the tape when durationEvent is false
            impreciseIconMargin: 3   // A tape and an icon are painted for imprecise instant
                                     // events. This attribute is the margin between the
                                     // bottom of the tape and the top of the icon in that
                                     // case.
    //        color:             "#58A0DC",
    //        impreciseColor:    "#58A0DC",
        },
        duration: {
            impreciseOpacity: 20 // tape opacity for imprecise part of duration events
      //      color:            "#58A0DC",
      //      impreciseColor:   "#58A0DC",
        },
        label: {
            backgroundOpacity: 50,// only used in detailed painter
               offsetFromLine:  3 // px left margin amount from icon's right edge
      //      backgroundColor:   "white",
      //      lineColor:         "#58A0DC",
        },
        highlightColors: [  // Use with getEventPainter().setHighlightMatcher
                            // See webapp/examples/examples.js
            "#FFFF00",
            "#FFC000",
            "#FF0000",
            "#0000FF"
        ],
        highlightLabelBackground: false, // When highlighting an event, also change the event's label background?
        bubble: {
            width:          250, // px
            maxHeight:        0, // px Maximum height of bubbles. 0 means no max height. 
                                 // scrollbar will be added for taller bubbles
            titleStyler: function(elmt) {
                elmt.className = "timeline-event-bubble-title";
            },
            bodyStyler: function(elmt) {
                elmt.className = "timeline-event-bubble-body";
            },
            imageStyler: function(elmt) {
                elmt.className = "timeline-event-bubble-image";
            },
            wikiStyler: function(elmt) {
                elmt.className = "timeline-event-bubble-wiki";
            },
            timeStyler: function(elmt) {
                elmt.className = "timeline-event-bubble-time";
            }
        }
    };
    
    this.mouseWheel = 'scroll'; // 'default', 'zoom', 'scroll'
};